fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/NewNumberParmBehaviorUtilTest.java

```diff
@@ -0,0 +1,467 @@
+        List<Object> expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        String expectedSql = "(Basic.NUMBER_VALUE > ? OR Basic.NUMBER_VALUE_HIGH > ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // lt - Less Than
+        queryParm             = generateParameter(SearchConstants.Prefix.LT, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE < ? OR Basic.NUMBER_VALUE_LOW < ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // ge - Greater than Equal
+        queryParm             = generateParameter(SearchConstants.Prefix.GE, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE >= ? OR Basic.NUMBER_VALUE_HIGH >= ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // le - Less than Equal
+        queryParm             = generateParameter(SearchConstants.Prefix.LE, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE <= ? OR Basic.NUMBER_VALUE_LOW <= ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // sa - starts after
+        queryParm             = generateParameter(SearchConstants.Prefix.SA, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        // expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "Basic.NUMBER_VALUE_LOW > ?";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // eb - Ends before
+        queryParm             = generateParameter(SearchConstants.Prefix.EB, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        // expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "Basic.NUMBER_VALUE_HIGH < ?";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+    }
+
+    @Test()
+    public void testPrecisionIntegerWithStartsAfter() throws FHIRPersistenceException {
+        // sa - starts after with integer
+        QueryParameter queryParm =
+                generateParameter(SearchConstants.Prefix.SA, null, "window-end", new String[] { "1" });
+        List<Object> expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1"));
+        // expectedBindVariables.add(new BigDecimal("1"));
```

Comment:
? what the comment for?

Reply:
The old code adds the same bind variable twice even though only one parameter marker is in the expression `MolecularSequence.NUMBER_VALUE_LOW > ?`. Commented out the extra while I was trying to work out what the intent was.